packages/richtext-lexical/src/packages/@lexical/markdown/index.ts

```diff
@@ -82,7 +82,7 @@ function $convertFromMarkdownString(
   transformers: Array<Transformer> = TRANSFORMERS,
   node?: ElementNode,
   shouldPreserveNewLines = false,
-  shouldMergeAdjacentLines = false,
+  shouldMergeAdjacentLines = true,
```

Comment:
This classifies as a bug fix rather than a breaking change. It is standard behavior and what most people will expect.